isic_archive/models/image.py

```diff
@@ -29,16 +28,26 @@ def initialize(self):
 
         self.prefixSearchFields = ['lowerName', 'name']
 
+    @staticmethod
+    def generateIsicID():
+        taken = True
+        id_: int = 0
+
+        while taken:
+            id_ = secrets.randbelow(10_000_000)
```

Comment:
If you think it makes it more obvious, we could encode the max ID using the same "7" digit places that the string-format uses, but it's up to you.

```suggestion
            id_ = secrets.randbelow(10**7)
```